services/QuillLMS/app/controllers/stripe_integration/subscription_checkout_sessions_controller.rb

```diff
@@ -29,7 +29,7 @@ def create
     end
 
     private def customer
-      User.find_by_stripe_customer_id_or_email(stripe_customer_id, customer_email)
+      User.find_by_stripe_customer_id_or_email!(stripe_customer_id, customer_email)
```

Comment:
You might consider using a specific `PurchaserNotFoundError` here as well for tracking (although I don't know if it's really possible to raise that error).

Reply:
Good idea.

Reply:
The more I think about it, this condition should be more or less symmetric with call laterwe should actually see this exception here first.